app/lib/database/QueryBuilder.js

```diff
@@ -23,14 +23,29 @@ const Utils = require("../Utils.js");
  * Class responsible for parsing url params, payloads of client request to sql queries
  */
 class QueryBuilder {
-    static build(params) {
+
+    static async prepareRunsDetectorsOneHotQuery(client) {
+        const prepare_func_name = 'prepare_runs_one_hot_query';
+        let command = `select ${prepare_func_name}();`;
+        try {
+            command = (await client.query(command)).rows[0][prepare_func_name];
+            return command;
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    static async build(params, client) {
         const matchParams = [];
         const excludeParams = [];
         const fromParams = [];
         const toParams = [];
 
-        for (const [key, value] of Object.entries(params)) {
+        for (let [key, value] of Object.entries(params)) {
             const queryParam = key.substring(0, key.lastIndexOf('-'));
+            if (Array.isArray(value)) {
+                value = value[0];
+            }
```

Comment:
I would have written it this way as it handles an empty array gracefully as well:
```js
value = Array.isArray(value) && value.length > 0 ? value.find(e => true) : value;
```